src/utils/ddb/ddb_commands.c

```diff
@@ -1275,3 +1283,170 @@ ddb_run_dev_replace(struct ddb_ctx *ctx, struct dev_replace_options *opt)
 
 	return rc;
 }
+
+enum { TIMESPEC_FMT_SIZE = sizeof("1970-01-01 00:00:00.000000000") };
+
+static int
+timespec2str(struct timespec *tspec, char *buf, size_t buf_size)
+{
+	struct tm date;
+	size_t    buf_len;
+	int       rc;
+
+	tzset();
+	if (localtime_r(&(tspec->tv_sec), &date) == NULL)
+		return d_errno2der(errno);
+
+	buf_len = strftime(buf, buf_size, "%F %T", &date);
+	if (buf_len == 0)
+		return -DER_INVAL;
+
+	rc = snprintf(&buf[buf_len], buf_size - buf_len, ".%09ld", tspec->tv_nsec);
+	if (rc < 0)
+		return -DER_INVAL;
+	if (rc >= buf_size - buf_len)
+		return -DER_TRUNC;
+
+	return -DER_SUCCESS;
+}
+
+static int
+dtx_print_time_stat(struct ddb_ctx *ctx, char *stat_name, uint64_t epoch, char *align)
+{
+	struct timespec tspec;
+	char            buf[TIMESPEC_FMT_SIZE];
+	int             rc;
+
+	rc = d_hlc2timespec(epoch, &tspec);
+	if (!SUCCESS(rc)) {
+		ddb_errorf(ctx, "Conversion error of the DTX stat %s: " DF_RC "\n", stat_name,
+			   DP_RC(rc));
+		return rc;
+	}
+	rc = timespec2str(&tspec, buf, TIMESPEC_FMT_SIZE);
+	if (!SUCCESS(rc)) {
+		ddb_errorf(ctx, "Conversion error of the DTX stat %s: " DF_RC "\n", stat_name,
+			   DP_RC(rc));
+		return rc;
+	}
+	ddb_printf(ctx, "\t- %s time:%s%s, %" PRIu64 "\n", stat_name, align, buf, epoch);
+
+	return -DER_SUCCESS;
+}
+
+struct dtx_stat_args {
+	struct ddb_ctx          *ctx;
+	struct dtx_stat_options *opt;
+	uint32_t                 cont_seen;
+	uint32_t                 cmt_cnt;
+};
+
+static int
+dtx_stat(struct ddb_ctx *ctx, struct dv_indexed_tree_path *itp, uint32_t *cnt)
+{
+	struct dtx_stat stat;
+	daos_handle_t   coh = {0};
+	uint32_t        cnt_tmp;
+	int             rc;
+
+	rc = dv_cont_open(ctx->dc_poh, itp_cont(itp), &coh);
+	if (!SUCCESS(rc))
+		goto done;
+
+	cnt_tmp = vos_dtx_get_cmt_cnt(coh);
```

Comment:
I would suggest `vos_dtx_get_cmt_cnt()` can return failure if hit corruption instead of triggering assertion.